src/ConfigurationClientManager.ts

```diff
@@ -0,0 +1,296 @@
+            await this.#discoverFallbackClients(host);
+        }
+
+        // Filter static clients whose backoff time has ended
+        let availableClients = this.#staticClients.filter(client => client.backoffEndTime <= currentTime);
+        // If there are dynamic clients, filter and concatenate them
+        if (this.#dynamicClients && this.#dynamicClients.length > 0) {
+            availableClients = availableClients.concat(
+                this.#dynamicClients
+                    .filter(client => client.backoffEndTime <= currentTime));
+        }
+
+        return availableClients;
+    }
+
+    async refreshClients() {
+        const currentTime = Date.now();
+        if (this.isFailoverable &&
+            currentTime > new Date(this.#lastFallbackClientRefreshAttempt + MINIMAL_CLIENT_REFRESH_INTERVAL).getTime()) {
+            this.#lastFallbackClientRefreshAttempt = currentTime;
+            const host = new URL(this.endpoint).hostname;
+            await this.#discoverFallbackClients(host);
+        }
+    }
+
+    async #discoverFallbackClients(host) {
+        let result;
+        try {
+            result = await Promise.race([
+                new Promise((_, reject) => setTimeout(() => reject(new Error("SRV record query timed out.")), SRV_QUERY_TIMEOUT)),
+                this.#querySrvTargetHost(host)
+            ]);
+        } catch (error) {
+            throw new Error(`Fail to build fallback clients, ${error.message}`);
+        }
+
+        const srvTargetHosts = result as string[];
+        // Shuffle the list of SRV target hosts
+        for (let i = srvTargetHosts.length - 1; i > 0; i--) {
+            const j = Math.floor(Math.random() * (i + 1));
+            [srvTargetHosts[i], srvTargetHosts[j]] = [srvTargetHosts[j], srvTargetHosts[i]];
+        }
+
+        const newDynamicClients: ConfigurationClientWrapper[] = [];
+        for (const host of srvTargetHosts) {
+            if (isValidEndpoint(host, this.#validDomain)) {
+                const targetEndpoint = `https://${host}`;
+                if (targetEndpoint.toLowerCase() === this.endpoint.toLowerCase()) {
+                    continue;
+                }
+                const client = this.#credential ? new AppConfigurationClient(targetEndpoint, this.#credential, this.#clientOptions) : new AppConfigurationClient(buildConnectionString(targetEndpoint, this.#secret, this.#id), this.#clientOptions);
+                newDynamicClients.push(new ConfigurationClientWrapper(targetEndpoint, client));
+            }
+        }
+
+        this.#dynamicClients = newDynamicClients;
+        this.#lastFallbackClientRefreshTime = Date.now();
+    }
+
+    #isFallbackClientDiscoveryDue(dateTime) {
+        return dateTime >= this.#lastFallbackClientRefreshAttempt + MINIMAL_CLIENT_REFRESH_INTERVAL
+            && (!this.#dynamicClients
+                || this.#dynamicClients.every(client => dateTime < client.backoffEndTime)
```

Comment:
Seems to be different from .NET provider's implementation [ref](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs#L224)

I forget the reason you told me why here is `<`. Could you explain it here in case I forget it again?

Reply:
I also didn't see this in .NET provider [ref](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs#L99)

Reply:
discussed offline

Reply:
Note: if there is any client with backoffEndTime <= Date.now(0 which means there is available client, then no need to discover more